### PR TITLE
Settings-store follow up from #8332

### PR DESCRIPTION
**Pull request by brusch**
See https://github.com/pimcore/pimcore/pull/8332#issuecomment-791243369

### CODE REVIEW

File: models/Tool/SettingsStore.php [hunk 1]
```diff
@@ -80,39 +80,40 @@ protected static function validateType(string $type): bool
...
-     * @param mixed $data
-     * @param string|null $scope
+     * @param int|string|bool|float $data
      * @param string $type
+     * @param string|null $scope
      * @return bool
      * @throws \Exception
      */
-    public static function set(string $id, $data, string $scope = null, string $type = 'string'): bool
+    public static function set(string $id, $data, string $type = 'string', string $scope = null): bool
```
Comment (jdreesen) [inline models/Tool/SettingsStore.php:89]:
```suggestion
    public static function set(string $id, $data, string $type = 'string', ?string $scope = null): bool
```

File: models/Tool/SettingsStore/Dao.php [hunk 2]
```diff
@@ -30,18 +30,18 @@ class Dao extends Model\Dao\AbstractDao
 
     /**
      * @param string $id
-     * @param mixed $data
+     * @param int|string|bool|float $data
      * @param string|null $scope
      * @param string $type
```
Comment (jdreesen) [inline models/Tool/SettingsStore/Dao.php:35]:
```suggestion
     * @param string $type
     * @param string|null $scope
```

File: models/Tool/SettingsStore/Dao.php [hunk 3]
```diff
@@ -30,18 +30,18 @@ class Dao extends Model\Dao\AbstractDao
...
      * @return bool
      */
-    public function set(string $id, $data, string $scope = null, string $type = 'string'): bool
+    public function set(string $id, $data, string $type = 'string', string $scope = null): bool
```
Comment (jdreesen) [inline models/Tool/SettingsStore/Dao.php:38]:
```suggestion
    public function set(string $id, $data, string $type = 'string', ?string $scope = null): bool
```

File: models/Tool/SettingsStore.php [hunk 4]
```diff
@@ -158,7 +159,7 @@ public function getScope(): ?string
      */
     public function setScope(?string $scope): void
```
Comment (jdreesen) [inline models/Tool/SettingsStore.php:160]: The `@param` annotation is wrong and must be `string|null` (can't comment on that line).

File: models/Tool/SettingsStore.php [hunk 5]
```diff
@@ -21,7 +21,7 @@
 use Pimcore\Model\Tool\SettingsStore\Dao;
 
 /**
- * @method Dao getById(string $id)
+ * @method Dao getById(string $id, ?string $scope)
```
Comment (jdreesen) [inline models/Tool/SettingsStore.php:24]: `Dao::getById()` actually sets the data on the model and then returns `bool`.
```suggestion
 * @method bool getById(string $id, ?string $scope)
```

**Comment by jdreesen** [comment]
What about adding some convenience/shortcut methods?

```php
    /**
     * @param string $id
     * @param bool $data
     * @param string|null $scope
     * @return bool
     * @throws \Exception
     */
    public static function setBoolean(string $id, bool $data, string $scope = null): bool
    {
        return self::set($id, $data, 'bool', $scope);
    }

    /**
     * @param string $id
     * @param int $data
     * @param string|null $scope
     * @return bool
     * @throws \Exception
     */
    public static function setInteger(string $id, int $data, string $scope = null): bool
    {
        return self::set($id, $data, 'int', $scope);
    }

    /**
     * @param string $id
     * @param float $data
     * @param string|null $scope
     * @return bool
     * @throws \Exception
     */
    public static function setFloat(string $id, float $data, string $scope = null): bool
    {
        return self::set($id, $data, 'float', $scope);
    }

    /**
     * @param string $id
     * @param string $data
     * @param string|null $scope
     * @return bool
     * @throws \Exception
     */
    public static function setString(string $id, string $data, string $scope = null): bool
    {
        return self::set($id, $data, 'string', $scope);
    }
```

**Comment by brusch** [comment]
@jdreesen I'd keep it simple for now 🤔

**Pull request merged**